pallets/bridge/src/lib.rs

```diff
@@ -278,47 +286,81 @@ pub mod pallet {
             Ok(())
         }
 
+        /// Register the specified currency_id
+        ///
+        /// Only registered currencies are allowed to cross-chain
+        ///
+        /// - `currency_id`: should be unique.
         #[pallet::weight(T::WeightInfo::register_currency())]
         #[transactional]
         pub fn register_currency(
             origin: OriginFor<T>,
             asset_id: AssetIdOf<T>,
-            currency_id: CurrencyId,
+            currency: Currency,
         ) -> DispatchResultWithPostInfo {
             Self::ensure_admin(origin)?;
 
             ensure!(
-                !CurrencyIds::<T>::contains_key(currency_id)
-                    && !AssetIds::<T>::contains_key(asset_id),
-                Error::<T>::CurrencyIdAlreadyRegistered,
+                !Currencies::<T>::contains_key(asset_id)
+                    && !AssetIds::<T>::contains_key(currency.clone().id),
+                Error::<T>::CurrencyAlreadyRegistered,
             );
 
-            CurrencyIds::<T>::insert(asset_id, currency_id);
-            AssetIds::<T>::insert(currency_id, asset_id);
+            Currencies::<T>::insert(asset_id, currency.clone());
+            AssetIds::<T>::insert(currency.id, asset_id);
 
-            Self::deposit_event(Event::CurrencyRegistered(asset_id, currency_id));
+            Self::deposit_event(Event::CurrencyRegistered(asset_id, currency.id));
             Ok(().into())
         }
 
+        /// Unregister the specified currency_id
         #[pallet::weight(T::WeightInfo::unregister_currency())]
         #[transactional]
         pub fn unregister_currency(
             origin: OriginFor<T>,
             currency_id: CurrencyId,
         ) -> DispatchResultWithPostInfo {
             Self::ensure_admin(origin)?;
-
             Self::ensure_currency_registered(currency_id)?;
 
             let asset_id = AssetIds::<T>::get(currency_id);
-            CurrencyIds::<T>::remove(asset_id);
+            Currencies::<T>::remove(asset_id);
             AssetIds::<T>::remove(currency_id);
 
             Self::deposit_event(Event::CurrencyRemoved(asset_id, currency_id));
             Ok(().into())
         }
 
+        /// Set the cross-chain transaction fee for a registered currency
+        #[pallet::weight(T::WeightInfo::set_currency_fee())]
+        #[transactional]
+        pub fn set_currency_fee(
+            origin: OriginFor<T>,
+            currency_id: CurrencyId,
+            new_fee: BalanceOf<T>,
+        ) -> DispatchResult {
+            Self::ensure_admin(origin)?;
+            Self::ensure_currency_registered(currency_id)?;
+
+            let asset_id = AssetIds::<T>::get(currency_id);
```

Comment:
Why not use currency_id? It confuses that getting an asset_id by currency_id and modify the Currencies by asset_id.

Reply:
*Reason*
the struct of currency is complex，it's not suitable as a key of maapping

*Solutions*
1. currency_id => asset_id => currency (Just like above👆)
2. currency_id => currency (need a extra storage)

Reply:
This did cause a bit confusion, maybe I should add another storage for `currency_id => currency`?